fip-6.md

```diff
@@ -0,0 +1,376 @@
+## Motivation
+Presently the FIO API provides a fixed set of locked token types which have been used at FIO genesis, the types of locks and associated characteristics of locks were implemented in code without the ability to configure the locks without re-coding the locking logic. We make a new data model for a lock which consists of a set of locking periods which are relative to the time of creation of the lock, and flags relating to the lock (votable, unlockable). These new locks do not relate to the "main net" locks used to provide tokens to FIO investors and employees at FIO genesis. New locks and old locks will all be processed with equal precedence according to the rules for the lock. :
+* These new locks will be used by the FIO foundation to sell or grant FIO tokens.
+* These can be used by users of the protocol to lock tokens within their own accounts (to protect from spending, and also to protect against hacking into accounts)
+* These new locks can be used by FIO users to transfer locked tokens to other users.
+* There should be maintenance calls provided to clean up state for the locks that are paid out.
+* Multiple locks may be specified on a single account as long as the locks do not exceed the balance of the account. When a lock is added to an account the unlocked balance is checked to determine if the lock is to be permitted.
+* The older type of locks used at main net genesis can co-exist with these new locks in accounts that already have a genesis token grant. both will be handled correctly.
+
+## Specification
+### Lock tokens
+#### New end point: *lock_tokens* 
+#### New action in new fio.lock contract locktokens
+##### Request
+When this request is processed, tokens will be transferred to the specified receiver public key from the actor account if these accounts are different, if they are the same, no transfer will take place, these tokens wil be locked, they can be voted if the lock specifies they are votable, they cannot be used for fees, or transferred until unlocked according to the specified schedule or unlocked (if the lock is unlockable). locked and unlocked funds may be intermixed within an accountthe unlocked funds will be usable as normal within the FIO protocol  (so fees can come from bundled transactions, and from these intermingled fees) . The accounting of all types locked tokens will be updated whenever transfer or voteproducer, or unlock are called by a user, so tokens will be unlocked whenever these calls are invoked. Users may have as many locks as they wish providing the sum of the locks does not exceed the balance of the account.
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|receiver_public_key|Yes|fio public key, see FIO public key validation rules.|FIO public key which will receive the locked tokens.|
+|votable|Yes|0 is not votable, 1 is votable.|This indicates if the locked amount is votable.|
+|unlockable|Yes|0 is not unlockable, 1 is unlockable.|This indicates if the lock is unlockable.|
```

Comment:
It doesn't say anywhere in this FIP what unlockable is used for. Is it so that I can unlock at anytime? If so what's the point of schedule in that case?

Reply:
it can be unlocked at any time, this overrides the schedule, if a lock is unlock able then the owner can unlock small subsets of funds for their own reasons, the schedule will still be adhered.

Reply:
Explain in FIP

Reply:
this has been added to the fip